src/hb/buffer.rs

```diff
@@ -1556,6 +1557,7 @@ pub const HB_BUFFER_SCRATCH_FLAG_SHAPER0: u32 = 0x01000000;
 // pub const HB_BUFFER_SCRATCH_FLAG_SHAPER3: u32 = 0x08000000;
 
 /// A buffer that contains an input string ready for shaping.
+#[cfg_attr(feature = "wasm-shaper", derive(Clone))]
```

Comment:
Why do you have to clone the buffer? This is a very expensive operation. Better avoid it.